api.planx.uk/modules/team/index.test.ts

```diff
@@ -0,0 +1,259 @@
+import supertest from "supertest";
+import app from "../../server";
+import { authHeader } from "../../tests/mockJWT";
+
+const mockAddMember = jest.fn();
+const mockRemoveMember = jest.fn();
+const mockChangeMemberRole = jest.fn();
+
+jest.mock("@opensystemslab/planx-core", () => {
+  return {
+    CoreDomainClient: jest.fn().mockImplementation(() => ({
+      team: {
+        addMember: () => mockAddMember(),
+        removeMember: () => mockRemoveMember(),
+        changeMemberRole: () => mockChangeMemberRole(),
+      },
+    })),
+  };
+});
+
+describe("Adding a user to a team", () => {
+  it("requires authentication", async () => {
+    await supertest(app)
+      .put("/team/123/add-member")
+      .send({
+        userId: 123,
+        role: "teamViewer",
+      })
+      .expect(401);
+  });
+
+  it("validates that userId is required", async () => {
+    await supertest(app)
+      .put("/team/123/add-member")
+      .set(authHeader())
+      .send({
+        role: "teamViewer",
+      })
+      .expect(400)
+      .then((res) => {
+        expect(res.body).toHaveProperty("issues");
+        expect(res.body).toHaveProperty("name", "ZodError");
+      });
+  });
+
+  it("validates that role is required", async () => {
+    await supertest(app)
+      .put("/team/123/add-member")
+      .set(authHeader())
+      .send({
+        userId: 123,
+      })
+      .expect(400)
+      .then((res) => {
+        expect(res.body).toHaveProperty("issues");
+        expect(res.body).toHaveProperty("name", "ZodError");
+      });
+  });
+
+  it("validates that role must one an accepted value", async () => {
```

Comment:
nit: typo here!

Reply:
Picked this up in the next PR - thank you